packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -115,35 +121,26 @@ const BindingConnector = ( {
 		[ attrName, onPropValueChange ]
 	);
 
-	useLayoutEffect( () => {
-		if ( typeof propValue !== 'undefined' ) {
-			updateBoundAttibute( propValue, attrValue );
-		} else if ( placeholder ) {
-			/*
-			 * Placeholder fallback.
-			 * If the attribute is `src` or `href`,
-			 * a placeholder can't be used because it is not a valid url.
-			 * Adding this workaround until
-			 * attributes and metadata fields types are improved and include `url`.
-			 */
-			const htmlAttribute =
-				getBlockType( blockName ).attributes[ attrName ].attribute;
-
-			if ( htmlAttribute === 'src' || htmlAttribute === 'href' ) {
-				updateBoundAttibute( null );
-				return;
-			}
+	if ( typeof propValue !== 'undefined' ) {
+		updateBoundAttibute( propValue, attrValue );
```

Comment:
Is it authorized in React to call "setState" in a render function? Isn't that something that can break easily or something. cc @jsnajdr

Reply:
Yes, it's possible and legal, you just need to be careful about not triggering infinite loops.

`setState` in a render function is how you can implement `getDerivedStateFromProps` in a functional component. If such a `setState` call happens, the output of the render function is guaranteed to be never committed to DOM (it's using props and state that are out of sync) and a second rerender is scheduled to be done immediately.

React FAQ mentions this technique for functional `getDerivedStateFromProps` here: https://legacy.reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops. Note how the `setState` functions are called directly in the `ScrollView` function, without any `useEffect`.

There's also a link to a "you probably don't need derived state" article that describes [some alternatives](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#preferred-solutions). But I think none of them match our case, because we're not working with local component state, but with a state in an external data store. Data store, when updated, also triggers a `setState`-ish operation being done on your component, so the process is similar, but not identical.

Reply:
Just to add to this: yes this is a legit react pattern for exactly the reason that Jarda explained: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes:~:text=adjust%20the%20state%20directly%20during%20rendering%3A

While we are indeed _using_ an external value (from a custom field) [here](https://github.com/WordPress/gutenberg/blob/1b91609f3ff5d7f293cd72c59c649c7f5b2f684a/packages/block-editor/src/hooks/use-bindings-attributes.js#L78), we're not setting it in store in any way. So this approach should be fine. 

That said, once we start implementing the ability to edit values, we'd have to use `useEffect()` somewhere because  we'd be setting a value in an external store.

Reply:
I see the Warning about rendering update the component state when rendering others:

```
react-dom.js?ver=18:73 Warning: Cannot update a component (`WithBlockBindingSupport(ParagraphBlock)`) while rendering a different component (`BindingConnector`). To locate the bad setState() call inside `BindingConnector`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
```

![image](https://github.com/WordPress/gutenberg/assets/77539/84c86722-1828-4209-a3bd-f58e00174237)

Could we ignore it?

Reply:
>  we're not setting it in store in any way. So this approach should be fine.

Is this something that we do not want to do at all?

Reply:
@kevin940726 and @talldan had several prototypes for Pattern Overrides that were trying to use the local copy of the `core/block-editor` store with some actions and selectors patched so everything works out of the box in React components. The biggest challenge when replicating the same idea on Block Bindings would be that for every selector run that returns the block attributes, there would be need to cross-check if the `metadata.bindings` is set and fetching the data from different sources. It is doable, but the concern would be the performance. I would be very happy to see that prototyped directly in `core/block-editor` store so we could measure the impact.